weasyl/commishinfo.py

```diff
@@ -10,8 +13,38 @@
 
 CURRENCY_PRECISION = 2
 
-
-def convert_currency(target):
+# map database charset to ISO4217 currency codes
+CURRENCY_CHARMAP = {
+    "": {"code": "USD", "name": "United States Dollar", "symbol": "&#36;"},
+    "e": {"code": "EUR", "name": "Euro", "symbol": "&#8364;"},
+    "p": {"code": "GBP", "name": "British Pound Sterling", "symbol": "&#163;"},
+    "y": {"code": "JPY", "name": "Japanese Yen", "symbol": "J&#165;"},
+    "c": {"code": "CAD", "name": "Canadian Dollar", "symbol": "C&#36;"},
+    "m": {"code": "MXN", "name": "Mexican Peso", "symbol": "M&#36;"},
+    "u": {"code": "AUD", "name": "Australian Dollar", "symbol": "A&#36;"},
+    "z": {"code": "NZD", "name": "New Zealand Dollar", "symbol": "NZ&#36;"},
+    "n": {"code": "CNY", "name": "Chinese Yuan", "symbol": "C&#165;"},
+    "f": {"code": "CHF", "name": "Swiss Franc", "symbol": "Fr"},
+}
+ALL_CURRENCIES = "".join([key for key, value in CURRENCY_CHARMAP.items()])
```

Comment:
This is `"".join(CURRENCY_CHARMAP.keys())`. Also – sorry for not noticing this in the earlier review – if it’s only used for `in` checks, `CURRENCY_CHARMAP` can be used directly (`"".join(c for c in stuff if c in CURRENCY_CHARMAP)`).